src/managers/ApplicationCommandPermissionsManager.js

```diff
@@ -0,0 +1,410 @@
+'use strict';
+
+const { Error, TypeError } = require('../errors');
+const Collection = require('../util/Collection');
+const { ApplicationCommandPermissionTypes, APIErrors } = require('../util/Constants');
+
+/**
+ * Manages API methods for permissions of Application Commands.
+ */
+class ApplicationCommandPermissionsManager {
+  constructor(manager) {
+    /**
+     * The manager or command that this manager belongs to
+     * @type {ApplicationCommandManager|ApplicationCommand}
+     */
+    this.manager = manager;
+    /**
+     * The guild that this manager acts on
+     * @type {?Guild}
+     */
+    this.guild = manager.guild ?? null;
+    /**
+     * The id of the guild that this manager acts on
+     * @type {?Snowflake}
+     */
+    this.guildID = manager.guildID ?? manager.guild?.id ?? null;
+    /**
+     * The id of the command this manager acts on
+     * @type {?Snowflake}
+     */
+    this.commandID = manager.id ?? null;
+    /**
+     * The client that instantiated this Manager
+     * @name ApplicationCommandPermissionsManager#client
+     * @type {Client}
+     * @readonly
+     */
+    Object.defineProperty(this, 'client', { value: manager.client });
+  }
+
+  /**
+   * The APIRouter path to the commands
+   * @param {Snowflake} guildID ID of the guild to use in the path,
+   * @param {Snowflake} [commandID] ID of the application command
+   * @returns {Object}
+   * @private
+   */
+  permissionsPath(guildID, commandID) {
+    let path = this.client.api.applications(this.client.application.id).guilds(guildID);
```

Comment:
How ESLint didn't complain, beats me:
```suggestion
    const path = this.client.api.applications(this.client.application.id).guilds(guildID);
```

Reply:
I think this eslint rule is actually missing from config 🙂